scripts/lib.sh

```diff
@@ -86,10 +86,11 @@ function await_cln_block_processing() {
 
 # Function for killing processes stored in FM_PID_FILE
 function kill_fedimint_processes {
+  $FM_LN1 plugin stop "gateway-cln-extension" 2>/dev/null || true;
```

Comment:
IMO our tests shouldn't call this. We should just let `lightningd` shut down our plugin. Running this command manually is helpful to see the `shutdown` handler actually run, though.

Reply:
True. We can remove this explicit plugin stop once we figure out how to be resilient against `lightningd` crashes.

Reply:
won't fix, for now

Reply:
https://github.com/fedimint/fedimint/issues/1766

Reply:
I guess that [`lightning-stop` is the proper command](https://lightning.readthedocs.io/lightning-stop.7.html) to shutdown clightning?

Reply:
That's correct. However, the intent here is to explicitly stop the plugin with `lightning-cli plugin stop` command, before we [shutdown `lightningd` alongside other processes we track in `$FM_PID_FILE`](https://github.com/fedimint/fedimint/blob/master/scripts/lib.sh#L90)

Reply:
Yes, and you can get to the same result when calling `lightning-stop` AFAIU. the stop command stops lightningd and that in turn send the shutdown hook to all the plugins, meaning that our plugins receives that hook and has the chance to shutdown within 30 seconds; otherwise it's killed by lightningd (see documentation).